### DISCUSSION TITLE
function proposal : ggshow_html()

### DISCUSSION THREAD

**Discussion by egayer** [JetBrains/lets-plot#1002] [Ideas]
As discussed with @alshan, here is an idea for a function that would allow interactive plotting in a default web browser (or any other). This would be interesting when you are not using an IDE that allows interactive plotting.

The function is very simple, it `ggsave` a temporary .html file and then open it in a web browser. It needs **webbrowser** and **tempfile** packages both included in The Python Standard Library.

```
from lets_plot import *
import webbrowser
import tempfile

def ggshow_html(p):
    """
    Shows a temporary interactive plot p in the default web browser. When using ggshow_html() for different plots,
    plots will appear in different tabs.
    This function can be tuned to call any other web browser installed (see example in the code).
    Eric Gayer 2022 - 2024
    
    Parameters
    ----------
    p : a Lets-Plot plot

    Returns
    -------
    Interactive plot in default web browser.

    """
    temp = tempfile.NamedTemporaryFile()
    try:
        ggsave(p, temp.name+'.html')
        
        ### Use your default web browser
        webbrowser.get().open('file://'+ temp.name +'.html', new=2)
        
        ### Use a different web browser, here Google Chrome on Mac OS
        #webbrowser.get('open -a /Applications/Google\ Chrome.app %s').open('file://'+ temp.name +'.html', new=2)
        
    finally:
        temp.close()
```

Example :

```
import numpy as np
from lets_plot import *

np.random.seed(12)
data = dict(
    cond=np.repeat(['A', 'B'], 200),
    rating=np.concatenate((np.random.normal(0, 1, 200), np.random.normal(1, 1.5, 200)))
    )

p= ggplot(data, aes(x='rating', fill='cond')) + ggsize(700, 300) + \
    geom_density(color='dark_green', alpha=.7) + scale_fill_brewer(type='seq') + \
    theme(panel_grid_major_x='blank')

ggshow_html(p)
```

**Comment by alshan** [comment]
Hi Eric, thank you for sharing! We will try to incorporate this code in some form into lets-plot.

**Comment by alshan** [comment]
Hi @egayer, we've added this feature.
In a nutshell: 
Instead of `LetsPlot.setup_html()` you will add [`LetsPlot.setup_show_ext()`](https://lets-plot.org/python/pages/api/lets_plot.LetsPlot.html#lets_plot.LetsPlot.setup_show_ext).
After that, `p.show()` will open the plot in a browser window.

**Reply by egayer** [reply]
HI @alshan, awesome ! This is very clever, I like the idea of setting the web browser of your choice at anytime.

I'll install the 4.3 and test it right away !

**Reply by egayer** [reply]
It works very well ! thanks you !
